src/pages/index.js

```diff
@@ -74,115 +88,266 @@ const closeModalOnEsc = (event) => {
 
 const closeModalOnOverlayClick = (event) => {
   const openedModal = document.querySelector(".modal_opened");
-  if (
-    openedModal &&
-    !openedModal
-      .querySelector(".modal__container, .modal__content")
-      .contains(event.target)
-  ) {
+  if (event.target.classList.contains("modal")) {
     closeModal(openedModal);
   }
```

Comment:
`  const openedModal = document.querySelector(".modal_opened");`


No need in searching the modal here because `event.target` is the needed modal: you find `modal` in its class list


```js

  if (event.target.classList.contains("modal")) {
    closeModal(event.target);
  }
```